src/rpc/blockchain.cpp

```diff
@@ -1166,6 +1166,11 @@ UniValue gettxoutsetinfo(const JSONRPCRequest& request)
         ret.push_back(Pair("hash_serialized_2", stats.hashSerialized.GetHex()));
         ret.push_back(Pair("disk_size", stats.nDiskSize));
         ret.push_back(Pair("total_amount", ValueFromAmount(stats.nTotalAmount)));
+        statsClient.gauge("utxoset.tx", stats.nTransactions, 1.0f);
```

Comment:
These make no sense here. This makes measurements dependent on calling an RPC, which IMHO is the wrong way to do it. Better solution would be to periodically (e.g. once every 5m) do the GetUTXOStats (in a background thread) and send metrics then.